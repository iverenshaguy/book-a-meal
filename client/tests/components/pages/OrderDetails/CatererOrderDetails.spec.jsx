import React from 'react';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import CatererOrderDetailsComponent from '../../../../src/components/pages/OrderDetails/CatererOrderDetails';
import CatererOrderDetailsContainer from '../../../../src/containers/pages/OrderDetails/CatererOrderDetails';
import { caterer, caterersOrdersObj, initialState } from '../../../setup/mockData';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore({
  ...initialState,
  auth: { ...initialState.auth, user: { ...initialState.auth.user, ...caterer } }
});

const { now } = Date;

describe('CatererOrderDetails', () => {
  beforeAll(() => {
    Date.now = jest.fn(() => 0);
  });

  afterAll(() => {
    Date.now = now;
  });

  it('should render correctly when not fetching', () => {
    const wrapper = shallow(<CatererOrderDetailsComponent
      fetchOrder={jest.fn()}
      deliverOrder={jest.fn()}
      order={caterersOrdersObj.orders[0]}
      delivering={false}
      match={{ params: { id: 'fb097bde-5959-45ff-8e21-51184fa60c25' } }}
    />);

    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('OrderPill')).toBeTruthy();
  });

  it('should render MiniPreloader when delivering an order', () => {
    const wrapper = shallow(<CatererOrderDetailsComponent
      fetchOrder={jest.fn()}
      deliverOrder={jest.fn()}
      order={caterersOrdersObj.orders[0]}
      delivering
      match={{ params: { id: 'fb097bde-5959-45ff-8e21-51184fa60c25' } }}
    />);

    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('MiniPreloader')).toBeTruthy();
  });

  it('should render message when an order doesn\'t exist', () => {
    const comp = (
      <Provider store={store}>
        <CatererOrderDetailsContainer
          order={null}
          match={{ params: { id: 'fb097bde-5959-45ff' } }}
        />
      </Provider>
    );

    const wrapper = mount(comp, rrcMock.get());

    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('p.text-center').text()).toEqual('This Order Does Not Exist');
  });

  it('should render a delivered order correctly', () => {
    const newStore = mockStore({
      ...initialState,
      singleOrder: { ...initialState.singleOrder, item: caterersOrdersObj.orders[0] },
      auth: { ...initialState.auth, user: { ...initialState.auth.user, ...caterer } }
    });

    const comp = (
      <Provider store={newStore}>
        <CatererOrderDetailsContainer
          match={{ params: { id: 'fb097bde-5959-45ff-8e21-51184fa60c25' } }}
        />
      </Provider>
    );

    const wrapper = mount(comp, rrcMock.get()).find(CatererOrderDetailsComponent);

    expect(wrapper.find('.success').text()).toEqual('Delivered');
  });

  it('should render a canceled order correctly', () => {
    const newStore = mockStore({
      ...initialState,
      singleOrder: { ...initialState.singleOrder, item: caterersOrdersObj.orders[3] },
      auth: { ...initialState.auth, user: { ...initialState.auth.user, ...caterer } }
    });

    const comp = (
      <Provider store={newStore}>
        <CatererOrderDetailsContainer
          match={{ params: { id: 'fb097bde-5959-45ff-8e21-51184fa60c35' } }}
        />
      </Provider>
    );

    const wrapper = mount(comp, rrcMock.get()).find(CatererOrderDetailsComponent);

    expect(wrapper.find('.danger').text()).toEqual('Canceled');
  });

  it('should render a pending order correctly and show a button to deliver the order', () => {
    const newStore = mockStore({
      ...initialState,
      singleOrder: { ...initialState.singleOrder, item: caterersOrdersObj.orders[2] },
      auth: { ...initialState.auth, user: { ...initialState.auth.user, ...caterer } }
    });

    const comp = (
      <Provider store={newStore}>
        <CatererOrderDetailsContainer
          match={{ params: { id: 'fb097bde-5959-45ff-8e21-51184fa60c26' } }}
        />
      </Provider>
    );

    const wrapper = mount(comp, rrcMock.get()).find(CatererOrderDetailsComponent);

    wrapper.find('button.warning').simulate('click');

    expect(wrapper.find('button.warning').text()).toEqual('Deliver');
  });

  it('should render the connected OrderDetails component correctly', () => {
    const newStore = mockStore({
      ...initialState,
      singleOrder: { ...initialState.singleOrder, item: caterersOrdersObj.orders[0] },
      auth: { ...initialState.auth, user: { ...initialState.auth.user, ...caterer } }
    });

    const comp = (
      <Provider store={newStore}>
        <CatererOrderDetailsContainer
          match={{ params: { id: 'fb097bde-5959-45ff-8e21-51184fa60c25' } }}
        />
      </Provider>
    );

    const wrapper = mount(comp, rrcMock.get());

    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });
});
