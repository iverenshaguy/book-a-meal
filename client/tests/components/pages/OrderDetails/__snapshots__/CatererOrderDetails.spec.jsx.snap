// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatererOrderDetails should render MiniPreloader when delivering an order 1`] = `
<Connect(View)
  showTime={true}
  type="orders"
>
  <div
    className="main-wrapper caterer"
  >
    <Link
      className="orders-back-link"
      replace={false}
      to="/orders"
    >
      ← Back To Orders
    </Link>
    <div
      className="order-confirmation order-details"
    >
      <h3
        className="text-center"
        style={
          Object {
            "paddingTop": "1em",
          }
        }
      >
        Order #
        fb097bde-5959-45ff-8e21-51184fa60c25
      </h3>
      <div
        className="text-center"
      >
        <MiniPreloader />
      </div>
    </div>
  </div>
</Connect(View)>
`;

exports[`CatererOrderDetails should render correctly when not fetching 1`] = `
<Connect(View)
  showTime={true}
  type="orders"
>
  <div
    className="main-wrapper caterer"
  >
    <Link
      className="orders-back-link"
      replace={false}
      to="/orders"
    >
      ← Back To Orders
    </Link>
    <div
      className="order-confirmation order-details"
    >
      <h3
        className="text-center"
        style={
          Object {
            "paddingTop": "1em",
          }
        }
      >
        Order #
        fb097bde-5959-45ff-8e21-51184fa60c25
      </h3>
      <div
        className="order-misc"
      >
        <p
          className="order-date text-center"
        >
          Friday, 6th April 2018 12:47 am
        </p>
        <p>
          Status: 
          <span
            className="success"
          >
            Delivered
          </span>
        </p>
        <p>
          Customer:
          Emiola Olasanmi
        </p>
        <p>
          Address Provided:
          4, Church Street, Yaba
        </p>
        <p>
          Number Provided:
          080xxxxxxxx
        </p>
      </div>
      <OrderSummary
        meals={
          Array [
            Object {
              "delivered": true,
              "description": "Sharwama contains no animal products, perfect for dieters",
              "id": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
              "imageUrl": "https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350",
              "price": "1200.00",
              "quantity": 1,
              "title": "Vegetable Sharwama and Guava Smoothie",
              "vegetarian": true,
            },
          ]
        }
        status="delivered"
      />
      <OrderAmount
        meals={
          Array [
            Object {
              "delivered": true,
              "description": "Sharwama contains no animal products, perfect for dieters",
              "id": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
              "imageUrl": "https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350",
              "price": "1200.00",
              "quantity": 1,
              "title": "Vegetable Sharwama and Guava Smoothie",
              "vegetarian": true,
            },
          ]
        }
        type="admin"
      />
    </div>
  </div>
</Connect(View)>
`;

exports[`CatererOrderDetails should render message when an order doesn't exist 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(CatererOrderDetails)
    match={
      Object {
        "params": Object {
          "id": "fb097bde-5959-45ff",
        },
      }
    }
    order={null}
  >
    <CatererOrderDetails
      deliverOrder={[Function]}
      delivering={false}
      fetchOrder={[Function]}
      match={
        Object {
          "params": Object {
            "id": "fb097bde-5959-45ff",
          },
        }
      }
      order={null}
    >
      <Connect(View)
        showTime={true}
        type="orders"
      >
        <View
          isFetching={false}
          logout={[Function]}
          showTime={true}
          type="orders"
          updateCurrentDate={null}
          uploading={false}
          user={
            Object {
              "address": "$, Herbert Macaulay Way, Yaba, Lagos",
              "businessName": "Food Circle",
              "email": "food@circle.com",
              "firstname": "Dave",
              "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
              "lastname": "Smith",
              "phoneNo": "08166557788",
              "role": "caterer",
            }
          }
        >
          <div
            className="admin"
          >
            <Header
              active="orders"
              dateType="orders"
              logout={[Function]}
              showTime={true}
              type="caterer"
              updateCurrentDate={null}
              user={
                Object {
                  "address": "$, Herbert Macaulay Way, Yaba, Lagos",
                  "businessName": "Food Circle",
                  "email": "food@circle.com",
                  "firstname": "Dave",
                  "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
                  "lastname": "Smith",
                  "phoneNo": "08166557788",
                  "role": "caterer",
                }
              }
            >
              <div
                className="header"
              >
                <div
                  className="navbar"
                >
                  <Connect(CatererHeader)
                    dateType="orders"
                    showTime={true}
                    updateCurrentDate={null}
                  >
                    <CatererHeader
                      currentDay="1970-01-01"
                      dateType="orders"
                      showTime={true}
                      toggleSideNav={[Function]}
                      updateCurrentDate={null}
                    >
                      <div
                        className="menu-control d-none-md"
                      >
                        <Link
                          className="d-none-md"
                          clickHandler={[Function]}
                          dataTip={null}
                          id={null}
                        >
                          <button
                            className="link-btn d-none-md"
                            data-tip={null}
                            id={null}
                            onClick={[Function]}
                            type="button"
                          >
                            <img
                              alt="toggler"
                              src="/img/toggler.png"
                            />
                          </button>
                        </Link>
                      </div>
                      <div
                        className="page-title d-none-md"
                      >
                        <h3>
                          <Link
                            replace={false}
                            to="/"
                          >
                            <a
                              href="/"
                              onClick={[Function]}
                            >
                              BOOK-A-MEAL
                            </a>
                          </Link>
                        </h3>
                      </div>
                      <div
                        className="page-title d-flex-md"
                      >
                        <div
                          className="date"
                        >
                          <h2>
                            Thursday, 1st January 1970
                            <span>
                              00:00
                            </span>
                          </h2>
                        </div>
                      </div>
                    </CatererHeader>
                  </Connect(CatererHeader)>
                </div>
              </div>
            </Header>
            <div
              className="content"
            >
              <Connect(SideNav)
                active="orders"
                logout={[Function]}
                user={
                  Object {
                    "address": "$, Herbert Macaulay Way, Yaba, Lagos",
                    "businessName": "Food Circle",
                    "email": "food@circle.com",
                    "firstname": "Dave",
                    "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
                    "lastname": "Smith",
                    "phoneNo": "08166557788",
                    "role": "caterer",
                  }
                }
              >
                <SideNav
                  active="orders"
                  logout={[Function]}
                  open={false}
                  push={[Function]}
                  toggleSideNav={[Function]}
                  user={
                    Object {
                      "address": "$, Herbert Macaulay Way, Yaba, Lagos",
                      "businessName": "Food Circle",
                      "email": "food@circle.com",
                      "firstname": "Dave",
                      "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
                      "lastname": "Smith",
                      "phoneNo": "08166557788",
                      "role": "caterer",
                    }
                  }
                >
                  <div
                    className="sidenav null"
                    id="sidenav"
                  >
                    <div
                      className="sidenav-content"
                    >
                      <div
                        className="sidenav-header"
                      >
                        <CloseIcon
                          btnID={null}
                          clickHandler={[Function]}
                          divClass="d-none-md"
                        >
                          <div
                            className="close d-none-md"
                          >
                            <Link
                              aria-hidden="true"
                              className={null}
                              clickHandler={[Function]}
                              dataTip={null}
                              id={null}
                            >
                              <button
                                className="link-btn null"
                                data-tip={null}
                                id={null}
                                onClick={[Function]}
                                type="button"
                              >
                                ×
                              </button>
                            </Link>
                          </div>
                        </CloseIcon>
                        <div
                          className="sidenav-title"
                        >
                          <h3
                            className="d-flex-md"
                          >
                            <Link
                              href="/"
                              replace={false}
                              to="/"
                            >
                              <a
                                href="/"
                                onClick={[Function]}
                              >
                                BOOK-A-MEAL
                              </a>
                            </Link>
                          </h3>
                          <div
                            className="username-circle"
                          >
                            <p>
                              F
                            </p>
                          </div>
                          <p>
                            Food Circle
                          </p>
                          <span>
                            <Link
                              className={null}
                              clickHandler={[Function]}
                              dataTip={null}
                              id={null}
                            >
                              <button
                                className="link-btn null"
                                data-tip={null}
                                id={null}
                                onClick={[Function]}
                                type="button"
                              >
                                Logout
                              </button>
                            </Link>
                          </span>
                        </div>
                      </div>
                      <div
                        className="sidenav-body"
                      >
                        <Link
                          className="menu-item false"
                          clickHandler={[Function]}
                          dataTip={null}
                          id={null}
                        >
                          <button
                            className="link-btn menu-item false"
                            data-tip={null}
                            id={null}
                            onClick={[Function]}
                            type="button"
                          >
                            Dashboard
                          </button>
                        </Link>
                        <Link
                          className="menu-item false"
                          clickHandler={[Function]}
                          dataTip={null}
                          id={null}
                        >
                          <button
                            className="link-btn menu-item false"
                            data-tip={null}
                            id={null}
                            onClick={[Function]}
                            type="button"
                          >
                            Meals
                          </button>
                        </Link>
                        <Link
                          className="menu-item false"
                          clickHandler={[Function]}
                          dataTip={null}
                          id={null}
                        >
                          <button
                            className="link-btn menu-item false"
                            data-tip={null}
                            id={null}
                            onClick={[Function]}
                            type="button"
                          >
                            Menu
                          </button>
                        </Link>
                        <Link
                          className="menu-item active"
                          clickHandler={[Function]}
                          dataTip={null}
                          id={null}
                        >
                          <button
                            className="link-btn menu-item active"
                            data-tip={null}
                            id={null}
                            onClick={[Function]}
                            type="button"
                          >
                            Orders
                          </button>
                        </Link>
                      </div>
                    </div>
                  </div>
                </SideNav>
              </Connect(SideNav)>
              <div
                className="content-wrapper"
                id="has-modal"
              >
                <p
                  className="text-center info"
                >
                  This Order Does Not Exist
                </p>
              </div>
            </div>
            <Connect(Modal)
              isFetching={false}
              logout={[Function]}
              showTime={true}
              type="orders"
              updateCurrentDate={null}
              uploading={false}
              user={
                Object {
                  "address": "$, Herbert Macaulay Way, Yaba, Lagos",
                  "businessName": "Food Circle",
                  "email": "food@circle.com",
                  "firstname": "Dave",
                  "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
                  "lastname": "Smith",
                  "phoneNo": "08166557788",
                  "role": "caterer",
                }
              }
            >
              <Modal
                isFetching={false}
                logout={[Function]}
                open={false}
                showTime={true}
                toggleModal={[Function]}
                type={null}
                updateCurrentDate={null}
                uploading={false}
                user={
                  Object {
                    "address": "$, Herbert Macaulay Way, Yaba, Lagos",
                    "businessName": "Food Circle",
                    "email": "food@circle.com",
                    "firstname": "Dave",
                    "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
                    "lastname": "Smith",
                    "phoneNo": "08166557788",
                    "role": "caterer",
                  }
                }
              />
            </Connect(Modal)>
            <Footer>
              <div
                className="footer"
              >
                <p>
                  © 2018. Iveren Shaguy. All images were gotten from the internet and are not mine.
                </p>
              </div>
            </Footer>
          </div>
        </View>
      </Connect(View)>
    </CatererOrderDetails>
  </Connect(CatererOrderDetails)>
</Provider>
`;

exports[`CatererOrderDetails should render the connected OrderDetails component correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(CatererOrderDetails)
    match={
      Object {
        "params": Object {
          "id": "fb097bde-5959-45ff-8e21-51184fa60c25",
        },
      }
    }
  >
    <CatererOrderDetails
      deliverOrder={[Function]}
      delivering={false}
      fetchOrder={[Function]}
      match={
        Object {
          "params": Object {
            "id": "fb097bde-5959-45ff-8e21-51184fa60c25",
          },
        }
      }
      order={
        Object {
          "createdAt": "2018-04-06T00:47:03.687Z",
          "customer": Object {
            "email": "emiola@olasanmi.com",
            "firstname": "Emiola",
            "lastname": "Olasanmi",
          },
          "deliveryAddress": "4, Church Street, Yaba",
          "deliveryPhoneNo": "080xxxxxxxx",
          "id": "fb097bde-5959-45ff-8e21-51184fa60c25",
          "meals": Array [
            Object {
              "delivered": true,
              "description": "Sharwama contains no animal products, perfect for dieters",
              "id": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
              "imageUrl": "https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350",
              "price": "1200.00",
              "quantity": 1,
              "title": "Vegetable Sharwama and Guava Smoothie",
              "vegetarian": true,
            },
          ],
          "status": "delivered",
          "updatedAt": "2018-04-06T00:47:03.687Z",
        }
      }
    >
      <Connect(View)
        showTime={true}
        type="orders"
      >
        <View
          isFetching={false}
          logout={[Function]}
          showTime={true}
          type="orders"
          updateCurrentDate={null}
          uploading={false}
          user={
            Object {
              "address": "$, Herbert Macaulay Way, Yaba, Lagos",
              "businessName": "Food Circle",
              "email": "food@circle.com",
              "firstname": "Dave",
              "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
              "lastname": "Smith",
              "phoneNo": "08166557788",
              "role": "caterer",
            }
          }
        >
          <div
            className="admin"
          >
            <Header
              active="orders"
              dateType="orders"
              logout={[Function]}
              showTime={true}
              type="caterer"
              updateCurrentDate={null}
              user={
                Object {
                  "address": "$, Herbert Macaulay Way, Yaba, Lagos",
                  "businessName": "Food Circle",
                  "email": "food@circle.com",
                  "firstname": "Dave",
                  "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
                  "lastname": "Smith",
                  "phoneNo": "08166557788",
                  "role": "caterer",
                }
              }
            >
              <div
                className="header"
              >
                <div
                  className="navbar"
                >
                  <Connect(CatererHeader)
                    dateType="orders"
                    showTime={true}
                    updateCurrentDate={null}
                  >
                    <CatererHeader
                      currentDay="1970-01-01"
                      dateType="orders"
                      showTime={true}
                      toggleSideNav={[Function]}
                      updateCurrentDate={null}
                    >
                      <div
                        className="menu-control d-none-md"
                      >
                        <Link
                          className="d-none-md"
                          clickHandler={[Function]}
                          dataTip={null}
                          id={null}
                        >
                          <button
                            className="link-btn d-none-md"
                            data-tip={null}
                            id={null}
                            onClick={[Function]}
                            type="button"
                          >
                            <img
                              alt="toggler"
                              src="/img/toggler.png"
                            />
                          </button>
                        </Link>
                      </div>
                      <div
                        className="page-title d-none-md"
                      >
                        <h3>
                          <Link
                            replace={false}
                            to="/"
                          >
                            <a
                              href="/"
                              onClick={[Function]}
                            >
                              BOOK-A-MEAL
                            </a>
                          </Link>
                        </h3>
                      </div>
                      <div
                        className="page-title d-flex-md"
                      >
                        <div
                          className="date"
                        >
                          <h2>
                            Thursday, 1st January 1970
                            <span>
                              00:00
                            </span>
                          </h2>
                        </div>
                      </div>
                    </CatererHeader>
                  </Connect(CatererHeader)>
                </div>
              </div>
            </Header>
            <div
              className="content"
            >
              <Connect(SideNav)
                active="orders"
                logout={[Function]}
                user={
                  Object {
                    "address": "$, Herbert Macaulay Way, Yaba, Lagos",
                    "businessName": "Food Circle",
                    "email": "food@circle.com",
                    "firstname": "Dave",
                    "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
                    "lastname": "Smith",
                    "phoneNo": "08166557788",
                    "role": "caterer",
                  }
                }
              >
                <SideNav
                  active="orders"
                  logout={[Function]}
                  open={false}
                  push={[Function]}
                  toggleSideNav={[Function]}
                  user={
                    Object {
                      "address": "$, Herbert Macaulay Way, Yaba, Lagos",
                      "businessName": "Food Circle",
                      "email": "food@circle.com",
                      "firstname": "Dave",
                      "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
                      "lastname": "Smith",
                      "phoneNo": "08166557788",
                      "role": "caterer",
                    }
                  }
                >
                  <div
                    className="sidenav null"
                    id="sidenav"
                  >
                    <div
                      className="sidenav-content"
                    >
                      <div
                        className="sidenav-header"
                      >
                        <CloseIcon
                          btnID={null}
                          clickHandler={[Function]}
                          divClass="d-none-md"
                        >
                          <div
                            className="close d-none-md"
                          >
                            <Link
                              aria-hidden="true"
                              className={null}
                              clickHandler={[Function]}
                              dataTip={null}
                              id={null}
                            >
                              <button
                                className="link-btn null"
                                data-tip={null}
                                id={null}
                                onClick={[Function]}
                                type="button"
                              >
                                ×
                              </button>
                            </Link>
                          </div>
                        </CloseIcon>
                        <div
                          className="sidenav-title"
                        >
                          <h3
                            className="d-flex-md"
                          >
                            <Link
                              href="/"
                              replace={false}
                              to="/"
                            >
                              <a
                                href="/"
                                onClick={[Function]}
                              >
                                BOOK-A-MEAL
                              </a>
                            </Link>
                          </h3>
                          <div
                            className="username-circle"
                          >
                            <p>
                              F
                            </p>
                          </div>
                          <p>
                            Food Circle
                          </p>
                          <span>
                            <Link
                              className={null}
                              clickHandler={[Function]}
                              dataTip={null}
                              id={null}
                            >
                              <button
                                className="link-btn null"
                                data-tip={null}
                                id={null}
                                onClick={[Function]}
                                type="button"
                              >
                                Logout
                              </button>
                            </Link>
                          </span>
                        </div>
                      </div>
                      <div
                        className="sidenav-body"
                      >
                        <Link
                          className="menu-item false"
                          clickHandler={[Function]}
                          dataTip={null}
                          id={null}
                        >
                          <button
                            className="link-btn menu-item false"
                            data-tip={null}
                            id={null}
                            onClick={[Function]}
                            type="button"
                          >
                            Dashboard
                          </button>
                        </Link>
                        <Link
                          className="menu-item false"
                          clickHandler={[Function]}
                          dataTip={null}
                          id={null}
                        >
                          <button
                            className="link-btn menu-item false"
                            data-tip={null}
                            id={null}
                            onClick={[Function]}
                            type="button"
                          >
                            Meals
                          </button>
                        </Link>
                        <Link
                          className="menu-item false"
                          clickHandler={[Function]}
                          dataTip={null}
                          id={null}
                        >
                          <button
                            className="link-btn menu-item false"
                            data-tip={null}
                            id={null}
                            onClick={[Function]}
                            type="button"
                          >
                            Menu
                          </button>
                        </Link>
                        <Link
                          className="menu-item active"
                          clickHandler={[Function]}
                          dataTip={null}
                          id={null}
                        >
                          <button
                            className="link-btn menu-item active"
                            data-tip={null}
                            id={null}
                            onClick={[Function]}
                            type="button"
                          >
                            Orders
                          </button>
                        </Link>
                      </div>
                    </div>
                  </div>
                </SideNav>
              </Connect(SideNav)>
              <div
                className="content-wrapper"
                id="has-modal"
              >
                <div
                  className="main-wrapper caterer"
                >
                  <Link
                    className="orders-back-link"
                    replace={false}
                    to="/orders"
                  >
                    <a
                      className="orders-back-link"
                      href="/orders"
                      onClick={[Function]}
                    >
                      ← Back To Orders
                    </a>
                  </Link>
                  <div
                    className="order-confirmation order-details"
                  >
                    <h3
                      className="text-center"
                      style={
                        Object {
                          "paddingTop": "1em",
                        }
                      }
                    >
                      Order #
                      fb097bde-5959-45ff-8e21-51184fa60c25
                    </h3>
                    <div
                      className="order-misc"
                    >
                      <p
                        className="order-date text-center"
                      >
                        Friday, 6th April 2018 12:47 am
                      </p>
                      <p>
                        Status: 
                        <span
                          className="success"
                        >
                          Delivered
                        </span>
                      </p>
                      <p>
                        Customer:
                        Emiola Olasanmi
                      </p>
                      <p>
                        Address Provided:
                        4, Church Street, Yaba
                      </p>
                      <p>
                        Number Provided:
                        080xxxxxxxx
                      </p>
                    </div>
                    <OrderSummary
                      meals={
                        Array [
                          Object {
                            "delivered": true,
                            "description": "Sharwama contains no animal products, perfect for dieters",
                            "id": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
                            "imageUrl": "https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350",
                            "price": "1200.00",
                            "quantity": 1,
                            "title": "Vegetable Sharwama and Guava Smoothie",
                            "vegetarian": true,
                          },
                        ]
                      }
                      status="delivered"
                    >
                      <div
                        className="order-summary"
                      >
                        <div
                          key="36d525d1-efc9-4b75-9999-3e3d8dc64ce3"
                        >
                          <p>
                            1
                            x
                          </p>
                          <p
                            style={
                              Object {
                                "paddingLeft": "5px",
                                "paddingRight": "5px",
                                "wordWrap": "break-word",
                              }
                            }
                          >
                            Vegetable Sharwama and Guava Smoothie
                          </p>
                          <p>
                            ₦
                            1200
                          </p>
                        </div>
                      </div>
                    </OrderSummary>
                    <OrderAmount
                      meals={
                        Array [
                          Object {
                            "delivered": true,
                            "description": "Sharwama contains no animal products, perfect for dieters",
                            "id": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
                            "imageUrl": "https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350",
                            "price": "1200.00",
                            "quantity": 1,
                            "title": "Vegetable Sharwama and Guava Smoothie",
                            "vegetarian": true,
                          },
                        ]
                      }
                      type="admin"
                    >
                      <div
                        className="order-amount admin-order-total"
                      >
                        <div>
                          <p>
                            Total
                          </p>
                          <h2>
                            ₦
                            1200
                          </h2>
                        </div>
                      </div>
                    </OrderAmount>
                  </div>
                </div>
              </div>
            </div>
            <Connect(Modal)
              isFetching={false}
              logout={[Function]}
              showTime={true}
              type="orders"
              updateCurrentDate={null}
              uploading={false}
              user={
                Object {
                  "address": "$, Herbert Macaulay Way, Yaba, Lagos",
                  "businessName": "Food Circle",
                  "email": "food@circle.com",
                  "firstname": "Dave",
                  "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
                  "lastname": "Smith",
                  "phoneNo": "08166557788",
                  "role": "caterer",
                }
              }
            >
              <Modal
                isFetching={false}
                logout={[Function]}
                open={false}
                showTime={true}
                toggleModal={[Function]}
                type={null}
                updateCurrentDate={null}
                uploading={false}
                user={
                  Object {
                    "address": "$, Herbert Macaulay Way, Yaba, Lagos",
                    "businessName": "Food Circle",
                    "email": "food@circle.com",
                    "firstname": "Dave",
                    "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
                    "lastname": "Smith",
                    "phoneNo": "08166557788",
                    "role": "caterer",
                  }
                }
              />
            </Connect(Modal)>
            <Footer>
              <div
                className="footer"
              >
                <p>
                  © 2018. Iveren Shaguy. All images were gotten from the internet and are not mine.
                </p>
              </div>
            </Footer>
          </div>
        </View>
      </Connect(View)>
    </CatererOrderDetails>
  </Connect(CatererOrderDetails)>
</Provider>
`;
